### PR TITLE
Tpl: add support for v1 of Apple App Site Association

### PR DESCRIPTION
**Pull request by chayev**
Closes: #216

### CODE REVIEW

File: src/hugolib/site_render.go [hunk 1]
```diff
@@ -357,7 +357,11 @@ func (s *Site) renderAASA() error {
 		return errors.New("failed to create targetPath for Apple App Site Associate")
 	}
 
-	templ := s.lookupLayouts("apple_app_site_association.json", "_default/apple_app_site_association.json", "_internal/_default/apple_app_site_association.json")
+	templ := s.lookupLayouts("apple_app_site_association_v2.json", "_default/apple_app_site_association_v2.json", "_internal/_default/apple_app_site_association_v2.json")
+
+	if s.Cfg.GetString("aasaVersion") == "v1" {
```
Comment (felicianotech) [inline src/hugolib/site_render.go:362]: This if block here needs to cover both cases. Right now you're loading the v2 template even when the user is requesting v1.

File: src/hugolib/aasa_test.go [hunk 2]
```diff
@@ -18,26 +18,43 @@ func TestAASAOutput(t *testing.T) {
 	testCases := []struct {
 		prefix  string
 		bundle  string
+		version string
 		passing bool
 	}{
 		{
 			"ABCDE12345",
 			"com.gothamhq.ios",
+			"",
```
Comment (felicianotech) [inline src/hugolib/aasa_test.go:27]: I don't think an empty string should be the value for the v2 schema when "v1" is the value for the v1 schema. We should use "v2" for the v2 schema to be consistent and explicit.

File: src/config/services/servicesConfig.go [hunk 3]
```diff
@@ -132,6 +134,7 @@ func DecodeConfig(cfg config.Provider) (c Config, err error) {
 	if c.AASA.Prefix == "" {
 		c.AASA.Prefix = cfg.GetString(aasaPrefixKey)
 		c.AASA.Bundle = cfg.GetString(aasaBundleKey)
+		c.AASA.Version = cfg.GetString(aasaVersion)
```
Comment (felicianotech) [inline src/config/services/servicesConfig.go:137]: This line may need to be in its own if block but I'm not sure. I'd say, as long as we have test cases covering multiple scenarios and they pass, we're probably fine to leave this here.

**Review by felicianotech — CHANGES_REQUESTED** [review]
1. I think the value for the version key should not include the letter "v". Here's examples of schemas that do this:

- https://snapcraft.io/docs/snapcraft-yaml-reference
- https://docs.docker.com/compose/compose-file/compose-file-v2/
- https://circleci.com/docs/2.0/configuration-reference/

Instead, the expected values can be a "1" or "2" that we parse as an integer, unless we expect Apple to go the SemVer route in the future, in which we do a string.

2. This PR should add this new feature to the features doc.
3. The default value for the version key should be set. I don't think an empty string is a good default here.

File: src/hugolib/site_render.go (hunk 1 above)
Comment (chayev) [reply src/hugolib/site_render.go]: Thanks for the feedback. I updated the logic to only set the v2 template if the aasaVersion is not set to 1

File: src/config/services/servicesConfig.go (hunk 3 above)
Comment (chayev) [reply src/config/services/servicesConfig.go]: I moved this to it's own block

File: src/hugolib/aasa_test.go [hunk 4]
```diff
@@ -18,26 +18,43 @@ func TestAASAOutput(t *testing.T) {
...
+			1,
+			true,
+		},
+		{
+			"",
+			"",
+			1,
 			false,
 		},
 	}
```
Comment (felicianotech) [inline src/hugolib/aasa_test.go:60]: Not supplying a version and relying on the default isn't tested here.

File: src/hugolib/site_render.go [hunk 5]
```diff
@@ -357,7 +357,13 @@ func (s *Site) renderAASA() error {
 		return errors.New("failed to create targetPath for Apple App Site Associate")
 	}
 
-	templ := s.lookupLayouts("apple_app_site_association.json", "_default/apple_app_site_association.json", "_internal/_default/apple_app_site_association.json")
+	var templ tpl.Template
+
+	if s.Cfg.GetInt("aasaVersion") == 1 {
+		templ = s.lookupLayouts("apple_app_site_association_v1.json", "_default/apple_app_site_association_v1.json", "_internal/_default/apple_app_site_association_v1.json")
```
Comment (felicianotech) [inline src/hugolib/site_render.go:363]: Might the file names be easier as `aasa_v1.json` and `aasa_v2.json`?

**Review by felicianotech — CHANGES_REQUESTED** [review]
Something is wrong with the tests. You've added a whole new file schema for AASA in this PR yet didn't touch the tests and they all pass.

File: src/hugolib/site_render.go (hunk 5 above)
Comment (chayev) [reply src/hugolib/site_render.go]: agreed! file names have been updated.

File: src/hugolib/aasa_test.go [hunk 6]
```diff
@@ -60,11 +60,17 @@ func TestAASAOutput(t *testing.T) {
 			depsCfg := deps.DepsCfg{Fs: fs, Cfg: cfg}
 
 			writeSourcesToSource(t, "content", fs, weightedSources...)
+
+			if tv == 9 {
+				// expect the site not to build
```
Comment (chayev) [inline src/hugolib/aasa_test.go:65]: @felicianotech - how can I test that `buildSingleSite(t, depsCfg, BuildCfg{})` fails or contains "* not a valid AASA version"?

Comment (felicianotech) [reply src/hugolib/aasa_test.go:65]: I don't know. It's a function. Follow it's return, the type, and what that struct and its methods do. I assume somewhere in there you'll find the answer to the question.

**Review by felicianotech — APPROVED** [review]

**Comment by felicianotech** [comment]
18 days later and it's done. Now AASA support should be complete.

**Pull request merged**